pubsub/pubsub.go

```diff
@@ -423,7 +423,15 @@ func (s *Subscription) Receive(ctx context.Context) (_ *Message, err error) {
 func (s *Subscription) getNextBatch(ctx context.Context, nMessages int) ([]*Message, error) {
 	var msgs []*driver.Message
 	for len(msgs) == 0 {
-		err := retry.Call(ctx, gax.Backoff{}, s.driver.IsRetryable, func() error {
+		// Allow unrecoverable errors from acks to be detected and reported.
+		s.mu.Lock()
+		err := s.err
+		s.mu.Unlock()
+		if s.err != nil {
```

Comment:
use `err` instead of `s.err` here and the line below.

Reply:
Done.

Reply:
Obsoleted by @vangent's idea above.